### PR TITLE
Make tokenQueue a plain slice

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title. This makes the code a bit more simple and avoids a few allocations. YAY?

## Benchmark

```
benchmark                                                                                       old ns/op     new ns/op     delta
BenchmarkUnmarshal/struct_data_v0.3-16                                                          17996         17630         -2.03%
BenchmarkUnmarshal/string_data_v0.3-16                                                          17431         17137         -1.69%
BenchmarkUnmarshal/data,_attributes_and_extensions_and_specversion_with_struct_data_v0.3-16     19363         18604         -3.92%
BenchmarkUnmarshal/base64_json_encoded_data_v1.0-16                                             12238         11992         -2.01%
BenchmarkUnmarshal/nil_data_v1.0-16                                                             16317         16048         -1.65%
BenchmarkUnmarshal/nil_data_v0.3-16                                                             16464         16268         -1.19%
BenchmarkUnmarshal/struct_data_v1.0-16                                                          17902         17714         -1.05%
BenchmarkUnmarshal/data,_attributes_and_extensions_and_specversion_with_struct_data_v1.0-16     19159         18351         -4.22%
BenchmarkUnmarshal/string_data_v1.0-16                                                          17243         17140         -0.60%
BenchmarkUnmarshal/base64_xml_encoded_data_v1.0-16                                              12775         12725         -0.39%

benchmark                                                                                       old allocs     new allocs     delta
BenchmarkUnmarshal/struct_data_v0.3-16                                                          79             78             -1.27%
BenchmarkUnmarshal/string_data_v0.3-16                                                          78             77             -1.28%
BenchmarkUnmarshal/data,_attributes_and_extensions_and_specversion_with_struct_data_v0.3-16     88             87             -1.14%
BenchmarkUnmarshal/base64_json_encoded_data_v1.0-16                                             52             51             -1.92%
BenchmarkUnmarshal/nil_data_v1.0-16                                                             74             73             -1.35%
BenchmarkUnmarshal/nil_data_v0.3-16                                                             74             73             -1.35%
BenchmarkUnmarshal/struct_data_v1.0-16                                                          77             76             -1.30%
BenchmarkUnmarshal/data,_attributes_and_extensions_and_specversion_with_struct_data_v1.0-16     88             87             -1.14%
BenchmarkUnmarshal/string_data_v1.0-16                                                          76             75             -1.32%
BenchmarkUnmarshal/base64_xml_encoded_data_v1.0-16                                              52             51             -1.92%

benchmark                                                                                       old bytes     new bytes     delta
BenchmarkUnmarshal/struct_data_v0.3-16                                                          5263          4750          -9.75%
BenchmarkUnmarshal/string_data_v0.3-16                                                          5247          4734          -9.78%
BenchmarkUnmarshal/data,_attributes_and_extensions_and_specversion_with_struct_data_v0.3-16     6537          6024          -7.85%
BenchmarkUnmarshal/base64_json_encoded_data_v1.0-16                                             4387          3874          -11.69%
BenchmarkUnmarshal/nil_data_v1.0-16                                                             5031          4518          -10.20%
BenchmarkUnmarshal/nil_data_v0.3-16                                                             5031          4518          -10.20%
BenchmarkUnmarshal/struct_data_v1.0-16                                                          5087          4574          -10.08%
BenchmarkUnmarshal/data,_attributes_and_extensions_and_specversion_with_struct_data_v1.0-16     6537          6024          -7.85%
BenchmarkUnmarshal/string_data_v1.0-16                                                          5071          4558          -10.12%
BenchmarkUnmarshal/base64_xml_encoded_data_v1.0-16                                              4403          3890          -11.65%

```

### CODE REVIEW

File: v2/event/event_unmarshal.go [hunk 1]
```diff
@@ -177,7 +174,10 @@ func ReadJson(out *Event, reader io.Reader) error {
 
 		// If no specversion, enqueue unconditionally
 		if !checkFlag(state, specVersionV03Flag|specVersionV1Flag) {
-			tokenQueue.push(key, iterator.ReadAny())
+			if tokenQueue == nil {
+				tokenQueue = make([]entry, 0, preallocQueueSize)
+			}
+			tokenQueue = append(tokenQueue, entry{key: key, value: iterator.ReadAny()})
```
Comment (slinkydeveloper) [inline v2/event/event_unmarshal.go:180]: I have to say that I'm very surprised! Using `append` is faster than assigning to preallocated memory with indexes?!

**Review by slinkydeveloper — COMMENTED** [review]
LGTM

File: v2/event/event_unmarshal.go (hunk 1 above)
Comment (markusthoemmes) [reply v2/event/event_unmarshal.go:180]: It's not. It should be equally quick. The win here is the saved allocations.

File: v2/event/event_unmarshal.go [hunk 2]
```diff
@@ -88,10 +85,10 @@ func ReadJson(out *Event, reader io.Reader) error {
 			// Now we have a specversion, so drain the token queue
 			switch eventContext := out.Context.(type) {
 			case *EventContextV03:
-				for i := 0; i < tokenQueue.i; i++ {
-					val := tokenQueue.slice[i].value
+				for _, entry := range tokenQueue {
```
Comment (slinkydeveloper) [inline v2/event/event_unmarshal.go:88]: ranging is faster than iterating on indexes? Can you bench with indexes?

Comment (markusthoemmes) [reply v2/event/event_unmarshal.go:88]: I did, no diff.

**Comment by slinkydeveloper** [comment]
@markusthoemmes can you fix the dco?

**Review by slinkydeveloper — APPROVED** [review]

**Pull request merged**